### PR TITLE
[V2] `select.styled` Grouped Feature

### PR DESCRIPTION
**Pull request by andrefelipe18**
### Checklist:

- [x] I read the [CONTRIBUTION GUIDE](https://tallstackui.com/docs/contribution)
- [x] I created a new branch on my fork for this pull request 
- [x] I ensure that the current tests are passing
- [no] I added tests that prove this pull request is working

### What:

- [x] Feature
- [ ] Enhancements
- [ ] Bugfix

### Description:

Adds new property **$group** to accept groups of values (https://primevue.org/select/#group)

### Demonstration & Notes:

<details>
  <summary>Blade</summary>

   ```blade
     <x-select.styled group :options="[
        [
            'label' => 'Brazil',
            'image' => 'https://placehold.co/15x15/cyan/white',
            'description' => 'Brazilian Real',
            'value' => [
                ['label' => 'São Paulo', 'value' => 4, 'image' => 'https://placehold.co/15x15/orange/white'],
                ['label' => 'Rio de Janeiro', 'value' => 5, 'image' => 'https://placehold.co/15x15/orange/white'],
                ['label' => 'Brasília', 'value' => 6, 'image' => 'https://placehold.co/15x15/orange/white']
            ]
        ],
        [
            'label' => 'Japan',
            'image' => 'https://placehold.co/15x15/yellow/white',
            'value' => [
                ['label' => 'Tokyo', 'value' => 1, 'image' => 'https://placehold.co/15x15/red/white'],
                ['label' => 'Kyoto', 'value' => 2, 'image' => 'https://placehold.co/15x15/red/white'],
                ['label' => 'Osaka', 'value' => 3, 'image' => 'https://placehold.co/15x15/red/white']
            ]
        ],
        [
            'label' => 'United States',
            'image' => 'https://placehold.co/15x15/purple/white',
            'value' => [
                ['label' => 'New York', 'value' => 7, 'image' => 'https://placehold.co/15x15/blue/white'],
                ['label' => 'Los Angeles', 'value' => 8, 'image' => 'https://placehold.co/15x15/blue/white'],
                ['label' => 'Chicago', 'value' => 9, 'image' => 'https://placehold.co/15x15/blue/white']
            ]
        ],
        [
            'label' => 'Germany',
            'image' => 'https://placehold.co/15x15/pink/white',
            'value' => [
                ['label' => 'Berlin', 'value' => 10, 'image' => 'https://placehold.co/15x15/green/white'],
                ['label' => 'Munich', 'value' => 11, 'image' => 'https://placehold.co/15x15/green/white'],
                ['label' => 'Hamburg', 'value' => 12, 'image' => 'https://placehold.co/15x15/green/white']
            ]
        ],
    ]" />
   ```
</details>

https://github.com/user-attachments/assets/eeae2bc9-d855-4f23-b307-1431b8725ba1

### CODE REVIEW

File: src/View/Components/Select/Styled.php [hunk 1]
```diff
@@ -39,6 +39,7 @@ public function __construct(
         public ?bool $required = false,
         public ?int $limit = null,
         public ?int $lazy = 10,
+        public ?bool $group = false,
```
Comment (devajmeireles) [inline src/View/Components/Select/Styled.php:42]: Rename this to `$grouped`

Comment (devajmeireles) [inline src/View/Components/Select/Styled.php]: Rename this to `$grouped`

File: src/resources/views/components/select/styled.blade.php [hunk 2]
```diff
@@ -127,6 +127,45 @@ class="w-full overflow-auto"
                         <x-tallstack-ui::icon.generic.loading @class($personalize['box.list.loading.class']) />
                     </div>
                 @endif
+                @if($group)
```
Comment (devajmeireles) [inline src/resources/views/components/select/styled.blade.php]:
```suggestion
                @if ($group)
```

**Review by devajmeireles — CHANGES_REQUESTED** [review]

**Comment by devajmeireles** [comment]
We're on the right track for that! Lastly, I'd love to see padding on all items in the group. Something very basic, like:

```
Brazil
  São Paulo
  Bahia
US
  California
  Texas
```

**Comment by devajmeireles** [comment]
Hey, @andrefelipe18 ! This looks amazing! Can you proceed with this PR? We only need one fix, I can handle all the other improvements that are needed.

We need to fix the parent-related text, because it is black, whether in light or dark mode:
CleanShot 2024-10-23 at 23 58 49

**Comment by andrefelipe18** [comment]
image

**Review by devajmeireles — APPROVED** [review]

**Review by devajmeireles — APPROVED** [review]

**Comment by devajmeireles** [comment]
Thanks, @andrefelipe18 !

**Pull request merged**